src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;img&gt; elements that supports file versioning.
+    /// </summary>
+    /// <remarks>
+    /// The tag helper won't process for cases with just the 'src' attribute.
+    /// </remarks>
+    [TargetElement("img", Attributes = FileVersionAttributeName)]
+    public class ImageTagHelper : TagHelper
+    {
+        private static readonly string Namespace = typeof(ImageTagHelper).Namespace;
+
+        private const string FileVersionAttributeName = "asp-file-version";
+        private const string SrcAttributeName = "src";
+
+        private FileVersionProvider _fileVersionProvider;
+        
+        /// <summary>
+        /// Src of the image.
+        /// </summary>
+        /// <remarks>
+        /// Passed through to the generated HTML in all cases.
+        /// </remarks>
+        [HtmlAttributeName(SrcAttributeName)]
+        public string Src { get; set; }
+
+        /// <summary>
+        /// Value indicating if file version should be appended to the src urls.
+        /// </summary>
+        /// <remarks>
+        /// If <c>true</c> then a query string "v" with the encoded content of the file is added.
+        /// </remarks>
+        [HtmlAttributeName(FileVersionAttributeName)]
+        public bool? FileVersion { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        protected internal ILogger<ImageTagHelper> Logger { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IHostingEnvironment HostingEnvironment { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public ViewContext ViewContext { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IMemoryCache Cache { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IHtmlEncoder HtmlEncoder { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IJavaScriptStringEncoder JavaScriptEncoder { get; set; }
+
+         /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            // Pass through attribute that is also a well-known HTML attribute.
```

Comment:
Entire method could be significantly simplified since `<link>` and `<script>` tag helpers only need `SetContent()` workaround due to globbing.  For example:
``` c#
public override void Process(TagHelperContext context, TagHelperOutput output)
{
    if (Src != null)
    {
        if (FileVersion == true)
        {
            EnsureFileVersionProvider();
            output.Attributes[SrcAttributeName] = _fileVersionProvider.AddFileVersionToPath(Src);
        }
        else
        {
            // Pass through attribute that is also a well-known HTML attribute.
            output.CopyHtmlAttribute(SrcAttributeName, context);
        }
    }
}
```